src/controllers/action-queue/types/pushCleaners.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+
+import { type Cleaner, asArray, asBoolean, asEither, asNumber, asObject, asOptional, asString, asValue } from 'cleaners'
+
+import { asBase64 } from '../../../util/cleaners/asBase64'
+import {
+  type AddressBalanceTrigger,
+  type BroadcastTx,
+  type NewPushEvent,
+  type PriceChangeTrigger,
+  type PriceLevelTrigger,
+  type PushEventState,
+  type PushMessage,
+  type PushTrigger,
+  type TxConfirmTrigger
+} from './pushTypes'
+
+export const asAddressBalanceTrigger: Cleaner<AddressBalanceTrigger> = asObject({
+  type: asValue('address-balance'),
+  pluginId: asString,
+  tokenId: asOptional(asString),
+  address: asString,
+  aboveAmount: asOptional(asString), // Satoshis or Wei or such
+  belowAmount: asOptional(asString) // Satoshis or Wei or such
+})
+
+export const asPriceChangeTrigger: Cleaner<PriceChangeTrigger> = asObject({
+  type: asValue('price-change'),
+  pluginId: asString,
+  tokenId: asOptional(asString),
+  dailyChange: asOptional(asNumber), // Percentage
+  hourlyChange: asOptional(asNumber) // Percentage
+})
+
+export const asPriceLevelTrigger: Cleaner<PriceLevelTrigger> = asObject({
+  type: asValue('price-level'),
+  currencyPair: asString, // From our rates server
+  aboveRate: asOptional(asNumber),
+  belowRate: asOptional(asNumber)
+})
+
+export const asTxConfirmTrigger: Cleaner<TxConfirmTrigger> = asObject({
+  type: asValue('tx-confirm'),
+  pluginId: asString,
+  confirmations: asNumber,
+  txid: asString
+})
+
+export const asPushTrigger: Cleaner<PushTrigger> = asEither(asAddressBalanceTrigger, asPriceChangeTrigger, asPriceLevelTrigger, asTxConfirmTrigger)
+
+export const asBroadcastTx: Cleaner<BroadcastTx> = asObject({
+  pluginId: asString,
+  rawTx: asBase64
+})
+
+export const asPushMessage: Cleaner<PushMessage> = asObject({
+  title: asOptional(asString),
+  body: asOptional(asString),
+  data: asOptional(asObject(asString))
+})
+
+export const asPushEventState: Cleaner<PushEventState> = asValue('waiting', 'cancelled', 'triggered', 'complete', 'hidden')
```

Comment:
I think based on your PR feedback we can get rid of `triggered`. This was a leftover from the days when we were going to have two separate processes, one to check prices and another to actually send the messages. Now the price checker just calls the broadcast method right away, so events never sit around `triggered` but un-broadcast anymore.

All the other states remain, but there is no way for you, the client, to actually get a `hidden` event with the way the API works.